axis_backend/apps/documents/services/document_service.py

```
@@ -0,0 +1,486 @@
+        if 'contract_id' in updates and 'client_id' in updates:
+            from apps.contracts.models import Contract
+            try:
+                contract = Contract.objects.get(id=updates['contract_id'])
+                if str(contract.client_id) != updates['client_id']:
+                    raise ValidationError({
+                        'contract_id': 'Contract does not belong to specified client'
+                    })
+            except Contract.DoesNotExist:
+                raise ValidationError({
+                    'contract_id': f'Contract with id {updates["contract_id"]} does not exist'
+                })
```

Ruff (0.14.5)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)